mongodbatlas/mongodbatlas.go

```diff
@@ -270,6 +314,14 @@ func (c *Client) OnRequestCompleted(rc RequestCompletionCallback) {
 func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) (*Response, error) {
 	resp, err := DoRequestWithClient(ctx, c.client, req)
```

Comment:
I would like to get rid of `DoRequestWithClient` since it doe snot offer any advantage but I don't know if someone is using it